src/components/shared/FilterNavigatorBar.js

```diff
@@ -29,34 +33,51 @@ class FilterNavigatorBar extends PureComponent<Props> {
   }
 
   render() {
-    const { className, items, selectedItem } = this.props;
+    const { className, items, selectedItem, profileSelection } = this.props;
     return (
-      <TransitionGroup
-        component="ol"
-        className={classNames('filterNavigatorBar', className)}
-      >
-        {items.map((item, i) => (
-          <CSSTransition
-            key={i}
-            classNames="filterNavigatorBarTransition"
-            timeout={300}
-          >
-            <li
-              data-index={i}
-              className={classNames('filterNavigatorBarItem', {
-                filterNavigatorBarRootItem: i === 0,
-                filterNavigatorBarBeforeSelectedItem: i === selectedItem - 1,
-                filterNavigatorBarSelectedItem: i === selectedItem,
-                filterNavigatorBarLeafItem: i === items.length - 1,
-              })}
-              title={item}
-              onClick={this._onLiClick}
+      <div className={classNames('filterNavigatorBar', className)}>
+        <TransitionGroup
+          component="ol"
+          className={classNames('filterNavigatorBar', className)}
+        >
+          {items.map((item, i) => (
+            <CSSTransition
+              key={i}
+              classNames="filterNavigatorBarTransition"
+              timeout={300}
             >
-              <span className="filterNavigatorBarItemContent">{item}</span>
-            </li>
-          </CSSTransition>
-        ))}
-      </TransitionGroup>
+              <li
+                data-index={i}
+                className={classNames('filterNavigatorBarItem', {
+                  filterNavigatorBarRootItem: i === 0,
+                  filterNavigatorBarBeforeSelectedItem: i === selectedItem - 1,
+                  filterNavigatorBarSelectedItem: i === selectedItem,
+                  filterNavigatorBarLeafItem: i === items.length - 1,
+                })}
+                title={item}
+                onClick={this._onLiClick}
+              >
+                <span className="filterNavigatorBarItemContent">{item}</span>
+              </li>
+            </CSSTransition>
+          ))}
+        </TransitionGroup>
+        {profileSelection.hasSelection &&
+        className === 'profileFilterNavigator' ? (
+          <span
+            className={classNames(
+              'filterNavigatorBarItem',
+              'filterNavigatorBarLeafItem'
+            )}
+          >
+            <span className="filterNavigatorBarItemContent">
+              {getFormattedTimeLength(
+                profileSelection.selectionEnd - profileSelection.selectionStart
+              )}
```

Comment:
As said above, this calculation should directly go to `ProfileFilterNavigator`. Here we should only display the `uncommitedItem` if present. No CSSTransition for this item ! :-)

Having a new class name would be a good idea too, so that we can add an opacity to make it appear paler.